From 07e9c6832863931b993b67801bf82d1f25845183 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Thu, 20 Jan 2022 12:16:26 +0000
Subject: [PATCH] Disable pkcs related tests

p11-kit is disabled by default in Photon Os gnutls package.

glib-networking assumes p11-kit is enabled in gnutls and tries to
run some pkcs related tests. This is not required.

If we add p11-kit to Photon repo, it ends up in a nasty circular dependency.

Let's dodge it for now by using this workaround.

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 tls/tests/certificate.c |  3 +++
 tls/tests/connection.c  |  2 ++
 tls/tests/meson.build   | 16 +---------------
 3 files changed, 6 insertions(+), 15 deletions(-)

diff --git a/tls/tests/certificate.c b/tls/tests/certificate.c
index 01c3c1a..b4db16f 100644
--- a/tls/tests/certificate.c
+++ b/tls/tests/certificate.c
@@ -811,8 +811,11 @@ main (int   argc,
   module_path = g_test_build_filename (G_TEST_BUILT, "mock-pkcs11.so", NULL);
   g_assert_true (g_file_test (module_path, G_FILE_TEST_EXISTS));
 
+#if 0
   g_assert (gnutls_pkcs11_init (GNUTLS_PKCS11_FLAG_MANUAL, NULL) == GNUTLS_E_SUCCESS);
   g_assert (gnutls_pkcs11_add_provider (module_path, NULL) == GNUTLS_E_SUCCESS);
+#endif
+
   g_free (module_path);
 #endif
 
diff --git a/tls/tests/connection.c b/tls/tests/connection.c
index 2127cbf..67ccd65 100644
--- a/tls/tests/connection.c
+++ b/tls/tests/connection.c
@@ -3062,10 +3062,12 @@ main (int   argc,
       module_path = g_strdup (spy_path);
     }
 
+#if 0
   ret = gnutls_pkcs11_init (GNUTLS_PKCS11_FLAG_MANUAL, NULL);
   g_assert_cmpint (ret, ==, GNUTLS_E_SUCCESS);
   ret = gnutls_pkcs11_add_provider (module_path, NULL);
   g_assert_cmpint (ret, ==, GNUTLS_E_SUCCESS);
+#endif
   g_free (module_path);
 #endif
 
diff --git a/tls/tests/meson.build b/tls/tests/meson.build
index 7415f91..e00413e 100644
--- a/tls/tests/meson.build
+++ b/tls/tests/meson.build
@@ -21,21 +21,7 @@ envs = [
   'G_TEST_BUILDDIR=' + meson.current_build_dir(),
 ]
 
-if backends.contains('gnutls')
-  mock_pkcs11_module = shared_module('mock-pkcs11',
-    sources: 'mock-pkcs11.c',
-    name_prefix: '',
-    gnu_symbol_visibility: 'hidden',
-    dependencies: [
-      gio_dep,
-      gnutls_dep,
-    ],
-    install: enable_installed_tests,
-    install_dir: installed_tests_execdir
-  )
-else
-  mock_pkcs11_module = []
-endif
+mock_pkcs11_module = []
 
 if enable_installed_tests
   install_subdir('files', install_dir: installed_tests_execdir)
-- 
2.31.1

